src/Microsoft.AspNet.SignalR.Redis/RedisMessageBus.cs

```diff
@@ -27,6 +27,7 @@ public class RedisMessageBus : ScaleoutMessageBus
         private string _connectionString;
         private int _state;
         private readonly object _callbackLock = new object();
+        private readonly SemaphoreSlim _redisConnectionEventLock = new SemaphoreSlim(1, 1);
```

Comment:
Wouldn't it mean that we would have to block on executing the Redis script though?